src/components/Nav/GlobalNav/SearchDetail.js

```diff
@@ -0,0 +1,237 @@
+import React, { useState } from 'react';
+import { BsSearch } from 'react-icons/bs';
+import styled from 'styled-components/macro';
+import Location from './modals/Location';
+import Calendar from './modals/Calendar';
+import Guest from './modals/Guest';
+import variables from '../../../styles/variables';
+import theme from '../../../styles/theme';
+
+const SearchDetail = ({
+  setIsOpenSearchBar,
+  modalRef,
+  onChange,
+  endDate,
+  startDate,
+  increaseGuestNum,
+  decreaseGuestNum,
+  guestNum,
+  setSelectedLocation,
+  selectedLocation,
+  dateToString,
+}) => {
+  const [currentId, setCurrentId] = useState(0);
```

Comment:
```suggestion
  const [currentId, setCurrentId] = useState(null);
```
null을 주고 currentId 하나로 아래에 있는 state들은 모두 없앨 수 있을 것 같아요.